storage-node/src/storage_manager.rs

```diff
@@ -5,23 +5,32 @@ use crate::{
     storage_reader::{s3::S3Reader, s3_diskmock::MockS3Reader, AsyncStorageReader},
 };
 
+use async_trait::async_trait;
 use bytes::Bytes;
 use log::debug;
 use parpulse_client::RequestParams;
 use tokio::sync::mpsc::Receiver;
 
+#[async_trait]
+pub trait StorageManager: Send + Sync {
```

Comment:
Why do we need to extract this out?

Reply:
to enable `route` to accept `StorageManager<C>` where C could be different